From d6d64eb6594bbffb7862ca352af592974580932a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?IOhannes=20m=20zm=C3=B6lnig?= <zmoelnig@iem.at>
Date: Thu, 30 Sep 2010 15:28:23 +0200
Subject: [PATCH 07/44] v4l2sink: Protect against NULL-pointer access

gst_v4l2sink_change_state() would free the pool without checking whether there
was a valid pool...
---
 sys/v4l2/gstv4l2sink.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/sys/v4l2/gstv4l2sink.c b/sys/v4l2/gstv4l2sink.c
index 679f7e2..7dc1382 100644
--- a/sys/v4l2/gstv4l2sink.c
+++ b/sys/v4l2/gstv4l2sink.c
@@ -461,7 +461,8 @@ gst_v4l2sink_change_state (GstElement * element, GstStateChange transition)
       }
       break;
     case GST_STATE_CHANGE_READY_TO_NULL:
-      gst_v4l2_buffer_pool_destroy (v4l2sink->pool);
+      if (NULL != v4l2sink->pool)
+        gst_v4l2_buffer_pool_destroy (v4l2sink->pool);
       v4l2sink->pool = NULL;
       /* close the device */
       if (!gst_v4l2_object_stop (v4l2sink->v4l2object))
-- 
1.7.1

